### PR TITLE
Redo submdspan

### PR DESCRIPTION
**Pull request by crtrott**

### CODE REVIEW

File: layout_padded/layout_padded.md [hunk 1]
```diff
@@ -2191,76 +2203,176 @@ template<class LayoutRightPaddedMapping>
...
-```c++
-  template<class Extents, std::size_t padding_stride, class... SliceSpecifiers>
-    constexpr auto submdspan_mapping(
-      const layout_left_padded<padding_stride>::template mapping<Extents>& src, 
-      SliceSpecifiers ... slices) -> /* see below */;
+<b>24.7.3.7.6 Specialization of `submdspan_mapping` [mdspan.submdspan.mapping]</b>
+
+<b>24.7.3.7.6.1 Common [mdspan.submdspan.mapping.common]</b>
+
+[1]{.pnum} The following elements apply to all functions in [mdspan.submdspan.impl].
```
Comment (mhoemmen) [inline layout_padded/layout_padded.md:2212]: `[mdspan.submdspan.impl]` doesn't exist, either here or in the Standard now.  Did you perhaps mean to write `[mdspan.submdspan.mapping]`?

```suggestion
[1]{.pnum} The following elements apply to all functions in [mdspan.submdspan.mapping].
```

File: layout_padded/layout_padded.md [hunk 2]
```diff
@@ -2191,76 +2203,176 @@ template<class LayoutRightPaddedMapping>
...
+
+   * [3.4]{.pnum} $S_k$ is a specialization of `strided_slice`
+
+[4]{.pnum} *Preconditions:* For each rank index `k` of `extents()`, all of the following are true:
+
+   * [4.1]{.pnum}  if $S_k$ is a specialization of `strided_slice`
+
+      * $s_k$`.extent` $= 0$, or
+
+      * $s_k$`.stride` $\gt 0$
```
Comment (mhoemmen) [inline layout_padded/layout_padded.md:2232]:
```suggestion
      * $s_k$`.stride` $\gt 0$; and
```

File: layout_padded/layout_padded.md [hunk 3]
```diff
@@ -2191,76 +2203,176 @@ template<class LayoutRightPaddedMapping>
...
+
+   * [4.2]{.pnum} $0 \le$ _`first`_`_<index_type, k>(slices...)` $\le$ _`last_<k>`_`(extents(), slices...)` $\le$ `extents.extent(k)`
```
Comment (mhoemmen) [inline layout_padded/layout_padded.md:2234]:
```suggestion
   * [4.2]{.pnum} $0 \le$ _`first`_`_<index_type, k>(slices...)` $\le$ _`last_<k>`_`(extents(), slices...)` $\le$ `extents.extent(k)`.
```

File: layout_padded/layout_padded.md [hunk 4]
```diff
@@ -2191,76 +2203,176 @@ template<class LayoutRightPaddedMapping>
...
+[6]{.pnum} Let `sub_strides` be an `array<SubExtents::index_type, SubExtents::rank()>` such that for each rank index `k` of `extents()` for which _`map-rank`_`[k]` is not `dynamic_extent`, `sub_strides[`_`map-rank`_`[k]]` equals:
+
+   * [6.1]{.pnum} `stride(k) * `_`de-ice`_`(`$s_k$`.stride)` if $S_k$ is a specialization of `strided_slice` and $s_k$`.stride` $\lt s_k$`.extent`;
+
+   * [6.2]{.pnum} otherwise, `stride(k)`.
+
+[7]{.pnum} Let `P`  be a parameter pack such that `is_same_v<make_index_sequence<rank()>, index_sequence<P...>>` is `true`.
+
+[8]{.pnum} Let `offset` be a value of type `size_t` equal to `(*this)(`_`first`_`_<index_type, P>(slices...)...)`.
+
```
Comment (mhoemmen) [inline layout_padded/layout_padded.md:2247]: Everything after this point diverges from the current Standard (the online draft includes `submdspan`; see https://eel.is/c++draft/views.multidim#mdspan.submdspan.submdspan ).

File: layout_padded/layout_padded.md [hunk 5]
```diff
@@ -2191,76 +2203,176 @@ template<class LayoutRightPaddedMapping>
...
+      * $S_0$ models _`index-pair-like`_`<index_type>`; and
 
-  * for `k` equal to `SubExtents::rank() - SubExtents::rank()`, `is_convertible_v<`$S_k$`, tuple<index_type, index_type>> || is_convertible_v<`$S_k$`, full_extent_t>` is `true`; otherwise
+      * for each `k` in the range $[$`1, SubExtents::rank()-1`$)$, `is_convertible_v<`$S_k$`, full_extent_t>` is `true`; and
 
-[9.6]{.pnum} `submdspan_mapping_result{layout_left_padded<dynamic_extent>::mapping(sub_ext, src.extent(0)), offset}` if
+      * for `k` equal to `SubExtents::rank()-1`, $S_k$ models _`index-pair-like`_`<index_type>` or `is_convertible_v<`$S_k$`, full_extent_t>` is `true`;
 
-  * `decltype(src)::layout_type` is a specialization of `layout_left_padded`; and
+   * [1.4]{.pnum} otherwise, `submdspan_mapping_result{layout_stride::mapping(sub_ext, sub_strides), offset}`.
```
Comment (mhoemmen) [inline layout_padded/layout_padded.md:2278]: Looks OK above here!

File: layout_padded/layout_padded.md (hunk 4 above)
Comment (crtrott) [reply layout_padded/layout_padded.md:2247]: yeah I know: because I rewrote it all

File: layout_padded/layout_padded.md [hunk 6]
```diff
@@ -2191,76 +2203,176 @@ template<class LayoutRightPaddedMapping>
...
+      <i>[Note: </i> If the above conditions are true, all $S_k$ with `k` $\lt$ `Extents::rank()-SubExtents::rank()` are convertible to `index_type`. <i>- end note]</i>
+
+   * [1.3]{.pnum} otherwise, `submdspan_mapping_result{layout_right_padded<Extents::static_extent(Extents::rank()-1)>::template mapping(sub_ext, extent(Extents::rank() - 1)), offset}` if
+
+      * for `k` equal to `Extents::rank() - 1` $S_k$ models _`index-pair-like`_`<index_type>`; and
+
+      * for each `k` in the range $[$`Extents::rank() - SubExtents::rank() + 1, Extents.rank() - 1`$)$, `is_convertible_v<`$S_k$`, full_extent_t>` is `true`; and
   
-  * for `k` equal to `Extents::rank() - 1` `is_convertible_v<` $S_k$ `, tuple<index_type, index_type>> || is_convertible_v<` $S_k$ `, full_extent_t>` is `true`; and
+      * for `k` equal to `Extents::rank()-SubExtents::rank()`, $S_k$ models _`index-pair-like`_`<index_type>` or `is_convertible_v<`$S_k$`, full_extent_t>` is `true`;
```
Comment (mhoemmen) [inline layout_padded/layout_padded.md:2307]: This is a whole range of `k` for `layout_left` -> `layout_left_padded`.  Here, it's just one `k`.

File: layout_padded/layout_padded.md [hunk 7]
```diff
@@ -2191,76 +2203,176 @@ template<class LayoutRightPaddedMapping>
...
+
+   * [1.4]{.pnum} otherwise, `submdspan_mapping_result{layout_stride::mapping(sub_ext, sub_strides), offset}`.
+
```
Comment (mhoemmen) [inline layout_padded/layout_padded.md:2310]: I've convinced myself that the `layout_left_padding` and `layout_right_padding` outcomes look correct.

File: layout_padded/layout_padded.md [hunk 8]
```diff
@@ -2191,76 +2203,176 @@ template<class LayoutRightPaddedMapping>
...
+   * [1.3]{.pnum} otherwise, `submdspan_mapping_result{layout_right_padded<padding_value>::mapping(sub_ext, stride(Extents::rank() - 1)), offset}` if
+
+      * for `k` equal to `Extents::rank() - 1`, $S_k$ models _`index-pair-like`_`<index_type>` or `is_convertible_v<`$S_0$`, full_extent_t>` is `true`; and
+
+      * for each `k` in the range $[$`Extents::rank() - SubExtents::rank() + 1, Extents.rank() - 1`$)$, `is_convertible_v<`$S_k$`, full_extent_t>` is `true`; and
+
+      * for `k` equal to `Extents::rank()-SubExtents::rank()`, $S_k$ models _`index-pair-like`_`<index_type>` or `is_convertible_v<`$S_k$`, full_extent_t>` is `true`;
+
+   * [1.4]{.pnum} otherwise, `submdspan_mapping_result{layout_stride::mapping(sub_ext, sub_strides), offset}`.
+
```
Comment (mhoemmen) [inline layout_padded/layout_padded.md:2376]: I've convinced myself that these two cases look correct.

File: layout_padded/layout_padded.md [hunk 9]
```diff
@@ -5,10 +5,10 @@ document: P2642
 date: 2023-07-13
```
Comment (mhoemmen) [inline layout_padded/layout_padded.md:5]:
```suggestion
date: 2023-10-13
```

**Pull request merged**